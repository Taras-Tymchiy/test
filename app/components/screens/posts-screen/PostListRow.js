// @flow
import * as React from 'react';
import { View } from 'react-native';
import { Post } from '../../../entities/Post';
import FormattedDate from '../../atoms/FormattedDate';
import ListRow from '../../atoms/ListRow';
import { TextPrimary, TextSecondary, TextRegular } from '../../atoms/Text';
import styles from './PostListRow.styles';

interface PostListRowProps {
    post: Post;
}

export default function PostListRow({post}: PostListRowProps) {
  const { id, created_at, user, text } = post;
  return (
    <ListRow>
      <View style={styles.header}> 
        <TextPrimary>{user.name}</TextPrimary>
        <TextSecondary>
          <FormattedDate value={created_at} />
        </TextSecondary>
      </View>
      <TextRegular>{text}</TextRegular>
    </ListRow>
  );
};
