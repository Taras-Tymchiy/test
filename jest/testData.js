import { Post } from "../app/entities/Post";


export const posts: Post[] = [
    {
        id: 1,
        id_str: '1',
        created_at: '2015-01-01',
        user: {
            id: 1,
            name: 'John Snow',
        },
        text: 'post text 1'
    },
    {
        id: 2,
        id_str: '2',
        created_at: '2015-01-02',
        user: {
            id: 2,
            name: 'Barack Obama',
        },
        text: 'post text 2'
    }
]