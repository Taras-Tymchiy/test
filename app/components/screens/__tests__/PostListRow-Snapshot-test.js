import React from 'react';
import renderer from 'react-test-renderer';
import PostListRow from '../posts-screen/PostListRow';
import { Text } from 'react-native';
import { posts } from '../../../../jest/testData';

it('renders a post in PostListRow using Snapshots', () => {
  expect(renderer.create(
    <PostListRow post={posts[0]} />
  )).toMatchSnapshot();
});

