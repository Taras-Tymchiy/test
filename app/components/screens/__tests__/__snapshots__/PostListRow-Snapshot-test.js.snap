// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a post in PostListRow using Snapshots 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "borderBottomWidth": 0.5,
        "borderColor": "#CCC",
        "padding": 12,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 6,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#222",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      John Snow
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#A0A0A0",
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        20/02/2018 15:01
      </Text>
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#222",
          "fontSize": 16,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    post text 1
  </Text>
</View>
`;
